Analysis file Enrico. Three plots:

enrico-leonardi opened the pull request:
1. %Diagnosis vs income class
2. %Medication (of diagnosed people) vs income class
3. %Diagnosis vs age

Commit:
Analysis file Enrico. Three plots:
1. Diagnosis vs income class
2. Medication (diagnosed people) vs income class
3. Diagnosis vs age

katmeld commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+
```
Where is the header?

katmeld commented on R/05_analysis_1.qmd:
````diff
@@ -0,0 +1,82 @@
+
+```{r}
+    #| message: false
+    library(dplyr)
````
Should have quotes around it "dplyr"

katmeld commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+    #| message: false
+    library(dplyr)
+    library("tidyverse")
+    rm(list = ls())
```
Clear environment before you load libraries for consistency

katmeld commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+    rm(list = ls())
+
+
+diabetes_data <- read_tsv(file = "../data/04_diabetes_data_augment.tsv")
```
Please add na = 'NA'

PernilleTJensen replied:
It should also be 03_diabetes_data_augment.tsv and not 03_diabetes_data_augment.tsv

PernilleTJensen replied:
Load the data in separate code chunk and add #| message:false

katmeld commented on R/05_analysis_1.qmd:
````diff
@@ -0,0 +1,82 @@
+```
+
+```{r}
+plot1 <- diabetes_data |>
````
the variable should be called something more meaningful. Maybe plot1_data?

katmeld commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+    y = "% of diabete-diagnosed",
+    caption = "To be read: within the class of income X, Y% of individuals got diagnosed with diabete."
+  )
+
```
The data should be piped into ggplot. Please have a line break after each comma

katmeld commented on R/05_analysis_1.qmd:
````diff
@@ -0,0 +1,82 @@
+
+
+```{r}
+plot2 <- diabetes_data |>
````
Variable name should be more meaningful

katmeld commented on R/05_analysis_1.qmd:
````diff
@@ -0,0 +1,82 @@
+```
+
+```{r}
+plot3 <- diabetes_data |>
````
Variable name should be more meaningful

katmeld commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+    x = "Age",
+    y = "% of diabete-diagnoses",
+    caption = "To be read: within the people of age X, Y% of individuals got diagnosed with diabete."
+  )
```
The data should be piped into ggplot. Please have a line break after each comma

katmeld reviewed (changes requested):
Please save your plots as pngs in the results folder. If there is not one, make one. Commit your plots as well

PernilleTJensen commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+                  filter(!is.na(income)) |>
+                  group_by(income) |>
+                  summarize(percentage = sum(diagnosis == 1)*100 / n())
+
```
filter(!is.na()) is Base R and not tidyverse.
Example of code you could use instead:
  diabetes_data |>
       select(income,
                diagnosis) |>
  drop_na() |>
  group_by(income) |>
  summarize(percentage = sum(diagnosis == 1)*100 / n())

PernilleTJensen commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+                  group_by(income) |>
+                  summarize(percentage = sum(diagnosis == 1)*100 / n())
+
+plot1 <- ggplot(plot1, aes(x = income, y = percentage, alpha = percentage)) +
```
Why alpha  = pencentage, it is the same as alpha = 0.5

PernilleTJensen commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+  theme(text = element_text(size=10), axis.text.x = element_text(angle = 15, vjust = 1, hjust=1)) +
+  coord_cartesian(ylim = c(0, 100)) +
+  labs(
+    title = "Diabete-diagnosed people vs Family income classes",
+    x = "Income classes",
+    y = "% of diabete-diagnosed",
+    caption = "To be read: within the class of income X, Y% of individuals got diagnosed with diabete."
```
For the title and y axis label it is called diabetes diagnosed and not diabete diagnosed.

Maybe is the caption not needed?

PernilleTJensen commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+                  filter(!is.na(income), diagnosis==1) |>
+                  group_by(income) |>
+                  summarize(percentage = sum(medication == 1)*100 / n())
+
```
filter(!is.na()) is Base R and not tidyverse.
Example of code you could use instead:
diabetes_data |>
select(income,
diagnosis) |>
drop_na() |>
group_by(income) |>
summarize(percentage = sum(diagnosis == 1)*100 / n())

PernilleTJensen commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+                  group_by(income) |>
+                  summarize(percentage = sum(medication == 1)*100 / n())
+
+plot2 <- ggplot(plot2, aes(x = income, y = percentage, alpha = percentage)) +
```
Why alpha = pencentage, it is the same as alpha = 0.5

PernilleTJensen commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+
+plot2 <- ggplot(plot2, aes(x = income, y = percentage, alpha = percentage)) +
+  geom_bar(stat="identity", color="black", fill='wheat', show.legend=FALSE) +
+  scale_alpha_continuous(range = c(0.2, 0.8)) +
```
Why also scale alpha, since it is already chosen?

PernilleTJensen commented on R/05_analysis_1.qmd:
````diff
@@ -0,0 +1,82 @@
+
+```{r}
+plot3 <- diabetes_data |>
+                  arrange(age) |>
+                  group_by(age) |>
````
arrange(age) and group_by(age) is doing the same thing so you dont need both

PernilleTJensen commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+  theme(text = element_text(size=10)) +
+  coord_cartesian(ylim = c(0, 100)) +
+  labs(
+    title = "Diagnosed people vs Age",
+    x = "Age",
+    y = "% of diabete-diagnoses",
+    caption = "To be read: within the people of age X, Y% of individuals got diagnosed with diabete."
```
For the title and y axis label it is called diabetes diagnosed and not diabete diagnosed.

Maybe is the caption not needed?

PernilleTJensen replied:
I think you should add the theme "theme_mininal2 so it is consistent with all the other plots in this project

PernilleTJensen commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+
+plot3 <- ggplot(plot3, aes(x = age, y = percentage, alpha = percentage)) +
+  geom_bar(stat="identity", color="black", fill="wheat", show.legend=FALSE) +
+  scale_alpha_continuous(range = c(0.2, 0.8)) +
```
Why also scale alpha, since it is already chosen?

enrico-leonardi replied:
I wanted alpha to vary with the level of percentage :) but not from 0 to 1 because otherwise some bars would have been blank: therefore I limited the opacity within 0.2 and 0.8. I'm editing all the rest btw

PernilleTJensen commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+                  group_by(age) |>
+                  summarize(percentage = sum(diagnosis == 1)*100 / n())
+
+plot3 <- ggplot(plot3, aes(x = age, y = percentage, alpha = percentage)) +
```
Why alpha = pencentage, it is the same as alpha = 0.5

PernilleTJensen commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+    caption = "To be read: within the class of income X, Y% of (diabete-diagnosed) individuals got medicated."
+  )
+
+plot2
```
I think you should add the theme "theme_mininal2 so it is consistent with all the other plots in this project

PernilleTJensen commented on R/05_analysis_1.qmd:
```diff
@@ -0,0 +1,82 @@
+  )
+
+
+plot1
```
I think you should add the theme "theme_mininal2 so it is consistent with all the other plots in this project

PernilleTJensen reviewed (commented):
-You are missing headers for all of you sections.
- rm(list=ls())) should not stand together with the libraries, it should be the first thing in the document.
-And the data load should also be in a seperate code chunck.
- The data is loaded from 04_diabetes_data_augment.tsv", which should be 03_diabetes_data_augment.tsv".
- The #| message:false should stand in the data load.
- Plot 1:
      -filter(!is.na()) is Base R and not tidyverse.
      - You should not put the plot1 into the ggplot but you should pipe it into the plot.
      - It does not make sense to mutate, you can just select income and diagnosis.
      - Example of code you could use instead:
      diabetes_data |>
  dplyr::select(income,
                diagnosis) |>
  drop_na() |>
  group_by(income) |>
  summarize(percentage = sum(diagnosis == 1)*100 / n())
  - You should also split one line into more lines ex. instead of ggplot(plot1, aes(x = income, y = percentage, alpha = percentage)) you should do
  plot1|> 
ggplot( aes(x = income, 
y = percentage, 
alpha = percentage))
-I does not make sense to say alpha = percentage, instead say alpha=0.5 + you dont need scale_alpha_continuous(range = c(0.2, 0.8)) 
- Maybe the caption is not needed?
- For the title and y axis label it is called diabetes diagnosed and not diabete diagnosed.

-All of the above things also applies for plot number 2.

-I think the themes should be theme_mininal so it is consisten with all the other plots in this project.

-All of the above thing also applies for plot 3 +
- arrange(age) and group_by(age) is doing the same thing so you dont need both.

Commit:
Don't mind this commit. Still editing my document

Commit:
Commit after merging

Merge remote-tracking branch 'origin/master' into Analysis_1
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

Commit:
Fixed mistakes in the code. Added the png plots.

katmeld committed:
removed unwanted files
